ui/raidboss/data/07-dt/dungeon/meso-terminal.ts

```diff
@@ -0,0 +1,405 @@
+const horizSouth: DirectionOutputIntercard[] = ['dirSW', 'dirSE'];
+const vertWest: DirectionOutputIntercard[] = ['dirNW', 'dirSW'];
+const vertEast: DirectionOutputIntercard[] = ['dirNE', 'dirSE'];
+
+const filterDirs = (
+  possSafe: DirectionOutputIntercard[],
+  knownUnsafe: DirectionOutputIntercard[],
+): DirectionOutputIntercard[] => {
+  return possSafe.filter((dir) => !knownUnsafe.includes(dir));
+};
+
+const findNorthTerror = (terrors: PluginCombatantState[]): PluginCombatantState | undefined => {
+  return terrors.filter((terror) => {
+    const terX = Math.round(Math.abs(terror.PosX));
+    const terY = Math.round(Math.abs(terror.PosY));
+    const x1 = 10;
+    const y1 = 13;
+    const x2 = 17;
+    const y2 = 12;
+    return (terX === x1 && terY === y1) || (terX === x2 && terY === y2);
+  })[0];
+};
+
+const triggerSet: TriggerSet<Data> = {
+  id: 'themesoterminal',
+  zoneId: ZoneId.TheMesoTerminal,
+  timelineFile: 'meso-terminal.txt',
+  initData: () => {
+    return {
+      chirurgeonSpots: [],
+      playerExecutionerId: undefined,
+      seenFirstStorm: false,
+      seenFirstBombardment: false,
+      impressionActive: false,
+      safeTerror: undefined,
+      kerauUnsafe: [],
+    };
+  },
+  triggers: [
+    {
+      id: 'Meso Terminal Chirurgeon Medicine Field',
+      type: 'StartsUsing',
+      netRegex: { id: 'AB16', source: 'Chirurgeon General', capture: false },
+      response: Responses.aoe(),
+    },
+    {
+      // The origin for the Chirurgeon is (270,12).
+      // Aerosol can be cast at four cardinal points,
+      // each ten units along an axis from the origin.
+      // (270,22), (280,12), (270,2),(260,12)
+      id: 'Meso Terminal Chirurgeon Pungent Aerosol',
+      type: 'StartsUsingExtra',
+      netRegex: { id: 'AB1F', capture: true },
+      infoText: (_data, matches, output) => {
+        const x = parseFloat(matches.x);
+        const y = parseFloat(matches.y);
+        const kbNum = Directions.xyTo4DirNum(x, y, 270, 12);
+        const safeDir = Directions.outputFromCardinalNum(kbNum);
+        return output.knockbackFrom!({ dir: output[safeDir]!() });
+      },
+      outputStrings: {
+        knockbackFrom: {
+          en: 'Knockback from ${dir}',
+        },
+        dirN: Outputs.north,
+        dirE: Outputs.east,
+        dirS: Outputs.south,
+        dirW: Outputs.west,
+        unknown: Outputs.unknown,
+      },
+    },
+    {
+      // AB1E is the big Sterile Sphere
+      // AB1D is small
+      id: 'Meso Terminal Chirurgeon Sterile Sphere Collect',
+      type: 'StartsUsingExtra',
+      netRegex: { id: 'AB1D', capture: true },
+      run: (data, matches) => {
+        const location = {
+          x: Math.round(parseFloat(matches.x)),
+          y: Math.round(parseFloat(matches.y)),
+        };
+        data.chirurgeonSpots.push(location);
+      },
+    },
+    {
+      id: 'Meso Terminal Chirurgeon Sterile Sphere Call',
```

Comment:
This might need some adjusting for Biochemical Front casts? On a first try of the trigger the output seemed to give a location that was being cleaved over by Biochemical Front.

Reply:
Can you expand on this? At no point does Biochemical Front overlap with Sterile Sphere, unless there's another section of the timeline I've seen.

Reply:
Unfortunately the order of Sterile/Pungent Spheres is random, so it's  tough to include that safely. Typically we've avoided doing "move to position" triggers for mechanics with clear indicators and 4.7s cast times. Do you think Biochemical Front moves fast enough to justify it even so?

Reply:
After a little more thinking about it I just added the trigger. It's not like it's that noisy.